crypto/generate.go

```diff
@@ -39,11 +39,17 @@ func GenerateSigningKeyPair(publicFileName, privateFileName, curveType string) (
 // GenerateHashPair generates a preimage and hash pair. This is useful in a commit reveal scheme such as what we use for anchor pre-commit > commit flow.
 func GenerateHashPair(preimageSize int) (preimage, hash []byte, err error) {
 	preimage = utils.RandomSlice(preimageSize)
+	hash, err = Sha256Hash(preimage)
+	return preimage, hash, err
+}
+
+// Sha256Hash wraps inconvenient sha256 hashing ops
+func Sha256Hash(value []byte) (hash []byte, err error) {
 	h := sha256.New()
-	_, err = h.Write(preimage)
+	_, err = h.Write(value)
 	if err != nil {
-		return []byte{}, []byte{}, err
+		return []byte{}, err
 	}
-	hash = h.Sum(hash)
-	return preimage, hash, nil
+	hash = h.Sum(nil)
```

Comment:
why is this line necessary if we are just appending nil?

Reply:
yes we can just make the following line
`return h.Sum(nil), nil`

Reply:
I mean, we can also just return the hash, no? Sum() does not change the underlying hash state, just appends the current hash to the parameter and returns.

Reply:
As far as I know there is no other function to get the underlying hash anyway, so we have to call this.

Reply:
okay, I see